Tests/MapboxNavigationTests/URLDataCacheTests.swift

```diff
@@ -0,0 +1,87 @@
+import XCTest
+import TestHelper
+@testable import MapboxNavigation
+
+class URLDataCacheTest: TestCase {
+    let url = ShieldImage.i280.baseURL
+    let cache: URLDataCache = URLDataCache()
+
+    override func setUp() {
+        super.setUp()
+        self.continueAfterFailure = false
+        cache.urlCache.diskCapacity = 0
+        cache.clearCache()
+    }
+    
+    private func exampleResponse(with storagePolicy: URLCache.StoragePolicy) -> CachedURLResponse {
+        let data = Fixture.JSONFromFileNamed(name: "sprite-info")
+        let response = URLResponse(url: url, mimeType: nil, expectedContentLength: data.count, textEncodingName: nil)
+        let cachedResponse = CachedURLResponse(response: response, data: data, storagePolicy: storagePolicy)
+        return cachedResponse
+    }
+    
+    private func urlRequest(with url: URL) -> URLRequest {
+        var request = URLRequest(url: url)
+        request.allHTTPHeaderFields = ["Accept": "image/*;q=0.8"]
+        request.cachePolicy = .reloadIgnoringCacheData
+        request.setValue(URLSession.userAgent, forHTTPHeaderField: "User-Agent")
+        return request
+    }
+
+    func testStoreCache() {
+        let response = exampleResponse(with: .allowed)
+        let request = urlRequest(with: url)
+        
+        cache.store(response, for: request)
+        
+        let cachedResponse = cache.response(for: request)
+        XCTAssertNotNil(cachedResponse)
+        XCTAssertEqual(cachedResponse, response)
+    }
+    
+    func testClearCache() {
+        let response = exampleResponse(with: .allowed)
+        let request = urlRequest(with: url)
+
+        cache.store(response, for: request)
+        XCTAssertEqual(cache.response(for: request), response)
+
+        cache.clearCache()
+        XCTAssertNil(cache.response(for: request)?.data)
+        XCTAssertEqual(cache.urlCache.currentMemoryUsage, 0)
+    }
+    
+    func testRemoveRequestCache() {
+        let response = exampleResponse(with: .allowed)
+        let request = urlRequest(with: url)
+        
+        cache.store(response, for: request)
+        XCTAssertNotNil(cache.response(for: request))
+        
+        cache.removeCache(for: request)
+        XCTAssertNil(cache.response(for: request)?.data)
+    }
+
+    func testStoreCacheInMemoryOnly() {
```

Comment:
We can also create stress test that generates response that is larger than allowed `memoryCapacity` and `diskCapacity` to verify that there are no leaks and cache behaves correctly.

Reply:
Thanks for the suggestion, the limit unit test is added.